homeassistant/components/kraken/sensor.py

```diff
@@ -0,0 +1,229 @@
+"""The kraken integration."""
+import logging
+from typing import Dict
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import async_entries_for_config_entry
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import KrakenData
+from .const import (
+    CONF_TRACKED_ASSET_PAIRS,
+    DISPATCH_CONFIG_UPDATED,
+    DOMAIN,
+    SENSOR_TYPES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add kraken entities from a config_entry."""
+
+    @callback
+    async def async_update_sensors(
+        hass: HomeAssistant, config_entry: ConfigEntry
+    ) -> None:
+        device_registry = await hass.helpers.device_registry.async_get_registry()
+
+        existing_devices = {
+            device.name: device.id
+            for device in async_entries_for_config_entry(
+                device_registry, config_entry.entry_id
+            )
+        }
+
+        for tracked_asset_pair in config_entry.options[CONF_TRACKED_ASSET_PAIRS]:
+            # Only create new devices
+            if create_device_name(tracked_asset_pair) in existing_devices.keys():
```

Comment:
Keys is the default.
```suggestion
            if create_device_name(tracked_asset_pair) in existing_devices:
```